src/server/scripts/DragonIsles/AberrusTheShadowedCrucible/boss_kazzara_the_hellforged.cpp

```diff
@@ -0,0 +1,158 @@
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "aberrus_the_shadowed_crucible.h"
+
+enum KazzaraSpells
+{
+    // Sundered NPCs
+    SPELL_FEAR                  = 220540,
+
+    // Kazzara
+    SPELL_DREAD_LANDING         = 411872,
+    SPELL_KAZZARA_INTRO         = 410541
+};
+
+enum KazzaraEvents
+{
+    EVENT_REMOVE_FLAGS          = 1
+};
+
+enum KazzaraIntroNpcs
+{
+    NPC_SUNDERED_EDGELORD       = 205734,
+    NPC_SUNDERED_PRESERVER      = 205735,
+    NPC_SUNDERED_SIEGEMASTER    = 205736,
+    NPC_SUNDERED_MANAWEAVER     = 205737,
+    NPC_SUNDERED_DEVASTATOR     = 198869,
+    NPC_SUNDERED_SCALEGUARD     = 198871,
+    NPC_SUNDERED_MANAWEAVER_2   = 198872,
+    NPC_SUNDERED_EDGELORD_2     = 198873,
+    NPC_SUNDERED_SIEGEMASTER_2  = 198874
+};
+
+class AllSunderedMobsInGrid
+{
+    public:
+        AllSunderedMobsInGrid() { }
+
+        bool operator()(Unit* u) const
+        {
+            if (!u->IsAlive() || u->IsInCombat())
+                return false;
+
+            switch (u->GetEntry())
+            {
+                case NPC_SUNDERED_EDGELORD:
+                case NPC_SUNDERED_PRESERVER:
+                case NPC_SUNDERED_SIEGEMASTER:
+                case NPC_SUNDERED_MANAWEAVER:
+                case NPC_SUNDERED_DEVASTATOR:
+                case NPC_SUNDERED_SCALEGUARD:
+                case NPC_SUNDERED_MANAWEAVER_2:
+                case NPC_SUNDERED_EDGELORD_2:
+                case NPC_SUNDERED_SIEGEMASTER_2:
+                    return true;
+                default:
+                    return false;
+            }
+
+            return true;
+        }
+};
+
+class CastFearEvent : public BasicEvent
+{
+    public:
+        CastFearEvent(Unit* caster) : _caster(caster) { }
+
+        bool Execute(uint64 /*execTime*/, uint32 /*diff*/) override
+        {
+            _caster->CastSpell(_caster, SPELL_FEAR, TRIGGERED_FULL_MASK);
+            return true;
+        }
+
+    private:
+        Unit* _caster;
+};
+
+// 201261 - Kazzara the Hellforged
+struct boss_kazzara_the_hellforged : public BossAI
+{
+    boss_kazzara_the_hellforged(Creature* creature) : BossAI(creature, DATA_KAZZARA_THE_HELLFORGED) { }
+
+    void JustAppeared() override
+    {
+        me->SetUnitFlag(UnitFlags(UNIT_FLAG_IMMUNE_TO_PC | UNIT_FLAG_IMMUNE_TO_NPC | UNIT_FLAG_UNINTERACTIBLE));
+        me->SetVisible(false);
+    }
+
+    void DoAction(int32 actionId) override
+    {
+        switch (actionId)
+        {
+            case ACTION_START_INTRO:
+            {
+                if (GameObject* gate = instance->GetGameObject(DATA_KAZZARA_GATE))
+                {
+                    gate->SetFlag(GO_FLAG_IN_USE);
+                    gate->SetGoState(GO_STATE_READY);
+                }
+
+                me->SetVisible(true);
+
+                DoCast(SPELL_DREAD_LANDING);
+                DoCast(SPELL_KAZZARA_INTRO);
+
+                std::vector<Creature*> sunderedMobs;
+                AllSunderedMobsInGrid check;
```

Comment:
instead of writing a custom grid searcher you could just give them a stringid in db (either in template or in creature, depends if the entry is used elsewhere in the instance) and then just use GetCreatureListWithOptionsInGrid

Reply:
they are not used for this kind of purporse so i gave them a stringid in creature_template